Libraries/Lists/VirtualizedList.js

```diff
@@ -1931,33 +2006,27 @@ type CellRendererProps = {
 };
 
 type CellRendererState = {
-  separatorProps: $ReadOnly<{|
-    highlighted: boolean,
-    leadingItem: ?Item,
-  |}>,
+  highlighted: boolean,
+  leadingItem: ?Item,
   ...
 };
 
-class CellRenderer extends React.Component<
+class CellRenderer extends React.PureComponent<
```

Comment:
This component accepts props which changes every render, so I think `PureComponent` checks will be false every time:
- `onLayout` - can be fixed
- `parentProps` - can't be fixed
- Maybe more?

Reply:
`onLayout` doesn't seem to cause extra renders in my manual tests. It's input value is given by the creator passed to `_pushCell()` function, which is only called when needed. So caching already is applied to `onLayout` too.

I think maybe for some use-cases there can be more optimization, for example not always everyone needs all the fields in parentProps, and sometimes the order and number of items in `data` doesn't change the internal of an item component, so maybe `ii` can be removed from `deps` too. I am interested to see other improvements done too.